File: app/src/main/java/com/kh69/passmath/data/cache/Question.java
package com.kh69.passmath.data.cache;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;


@Entity
public class Question {
    @PrimaryKey
    @SerializedName("_id")
    @Expose
    private String id;

    @SerializedName("qtn_text")
    @Expose
    private String text;

    @SerializedName("year")
    @Expose
    private int year;

    @SerializedName("paper")
    @Expose
    private int paper;

    @SerializedName("section")
    @Expose
    private String section;

    @SerializedName("topic")
    @Expose
    private String topic;

    @SerializedName("answer")
    @Expose
    private String answer;

    @SerializedName("katex_question")
    @Expose
    private String katex_question;

    @SerializedName("katex_answer")
    @Expose
    private String katex_answer;

    @SerializedName("edited")
    @Expose
    private boolean edited;

    public Question(String text, int year, int paper, String section, String topic, String answer, String katex_question, String katex_answer, boolean edited) {
        this.text           = text;
        this.year           = year;
        this.paper          = paper;
        this.section        = section;
        this.topic          = topic;
        this.answer         = answer;
        this.katex_question = katex_question;
        this.katex_answer   = katex_answer;
        this.edited         = edited;
    }

    public Question(String id, String text, int year, int paper, String section, String topic, String answer, String katex_question, String katex_answer, boolean edited) {
        this.id             = id;
        this.text           = text;
        this.year           = year;
        this.paper          = paper;
        this.section        = section;
        this.topic          = topic;
        this.answer         = answer;
        this.katex_question = katex_question;
        this.katex_answer   = katex_answer;
        this.edited         = edited;
    }

    public boolean getEdited() {
        return edited;
    }

    public void setEdited(boolean edited) {
        this.edited = edited;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getPaper() {
        return paper;
    }

    public void setPaper(int paper) {
        this.paper = paper;
    }

    public String getSection() {
        return section;
    }

    public void setSection(String section) {
        this.section = section;
    }

    public String getTopic() {
        return topic;
    }

    public void setTopic(String topic) {
        this.topic = topic;
    }

    public String getAnswer() {
        return answer;
    }

    public void setAnswer(String answer) {
        this.answer = answer;
    }

    public String getKatex_question() {
        return katex_question;
    }

    public void setKatex_question(String katex_question) {
        this.katex_question = katex_question;
    }

    public String getKatex_answer() {
        return katex_answer;
    }

    public void setKatex_answer(String katex_answer) {
        this.katex_answer = katex_answer;
    }
}

File: app/src/main/java/com/kh69/passmath/features/Question/domain/model/Question.kt
package com.kh69.passmath.features.question.domain.model

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "questions")
data class Question(
    @PrimaryKey
    @ColumnInfo(name = "questionId")
    @SerializedName("_id")
    @Expose
    @NonNull
    val id: String,

    @SerializedName("qtn_text")
    @Expose
    val text: String,

    @SerializedName("year")
    @Expose
    val year: Int,

    @SerializedName("paper")
    @Expose
    val paper: Int,

    @SerializedName("section")
    @Expose
    val section: String,

    @SerializedName("topic")
    @Expose
    val topic: String,

    @SerializedName("answer")
    @Expose
    val answer: String,

    @SerializedName("katex_question")
    @Expose
    val katex_question: String,

    @SerializedName("katex_answer")
    @Expose
    val katex_answer: String,

    @SerializedName("edited")
    @Expose
    val edited: Boolean,

    )

class InvalidNoteException(message: String) : Exception(message)